I think we should set pysaml2 to only build darwin on python38. 

```
            handler.flush()                                                                                                                                                                             handler.close()                                                                                                                                                                                                                                                                                                                                                                     print(c.logger)                                                                                                                                                                     >       c.setup_logger()                                                                                                                                                                                                                                                                                                                                                                tests/test_31_config.py:331:                                                                                                                                                                _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _                                                                                                             /nix/store/cgqmjmh17s54yn3m3rz8ivhzkv5gvgiq-python3.7-pysaml2-5.0.0/lib/python3.7/site-packages/saml2/config.py:509: in setup_logger                                                            root_logger.addHandler(self.log_handler())                                                                                                                                              /nix/store/cgqmjmh17s54yn3m3rz8ivhzkv5gvgiq-python3.7-pysaml2-5.0.0/lib/python3.7/site-packages/saml2/config.py:476: in log_handler                                                             handler = LOG_HANDLER[htyp](**args)                                                                                                                                                     /nix/store/i2fhj70arnf7d8h2scz5kmvwdi9809q4-python3-3.7.9/lib/python3.7/logging/handlers.py:829: in __init__                                                                                    ress = socket.getaddrinfo(host, port, 0, socktype)                                                                                                                                      _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _                                                                                                                                                                                                                                                                                                         host = 'localhost', port = 514, family = 0, type = <SocketKind.SOCK_DGRAM: 2>                                                                                                               proto = 0, flags = 0                                                                                                                                                                                                                                                                                                                                                                        def getaddrinfo(host, port, family=0, type=0, proto=0, flags=0):
        """Resolve host and port into list of address info entries.

        Translate the host/port argument into a sequence of 5-tuples that contain
        all the necessary arguments for creating a socket connected to that service.
        host is a domain name, a string representation of an IPv4/v6 address or
        None. port is a string service name such as 'http', a numeric port number or
        None. By passing None as the value of host and port, you can pass NULL to
        the underlying C API.

        The family, type and proto arguments can be optionally specified in order to
        narrow the list of addresses returned. Passing zero as a value for each of
        these arguments selects the full range of results.
        """
        # We override this function since we want to translate the numeric family
        # and socket type values to enum constants.
        addrlist = []
>       for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
E       socket.gaierror: [Errno 8] nodename nor servname provided, or not known

/nix/store/i2fhj70arnf7d8h2scz5kmvwdi9809q4-python3-3.7.9/lib/python3.7/socket.py:752: gaierror
```

--- a/pkgs/development/libraries/xmlsec/default.nix
+++ b/pkgs/development/libraries/xmlsec/default.nix
@@ -16,8 +16,7 @@ stdenv.mkDerivation {
 
   patches = [
     ./lt_dladdsearchdir.patch
-    ./remove_bsd_base64_decode_flag.patch
-  ];
+  ] ++ stdenv.lib.optionals stdenv.isDarwin [ ./remove_bsd_base64_decode_flag.patch ];
   postPatch = ''
     substituteAllInPlace src/dl.c
   '';

